### PR TITLE
LP-2279 Fix all linting issues of philu_api

### PR DESCRIPTION
**Pull request by monofal**
### Description

[LP-2279](https://philanthropyu.atlassian.net/browse/LP-2279)

### Notes
I have suppressed borad-except warning from pylint on couple of places. The reason being they were placed there to catch all exceptions and logged the errors. As per pep8

** A good rule of thumb is to limit use of bare 'except' clauses to two cases:

1. If the exception handler will be printing out or logging the traceback; at least the user will be aware that an error has occurred.
2. If the code needs to do some cleanup work, but then lets the exception propagate upwards with raise. try...finally can be a better way to handle this case.

### Final report:
```
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### CODE REVIEW

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: lms/djangoapps/philu_api/views.py [hunk 1]
```diff
@@ -204,8 +215,8 @@ def get_user_chat(request):
     username = request.user.username
     headers = {'Authorization': 'Bearer ' + settings.NODEBB_MASTER_TOKEN}
     response = requests.post(chat_endpoint,
-        data={'_uid': 1, 'username': username},
-        headers=headers)
+                             data={'_uid': 1, 'username': username},
+                             headers=headers)
```
[the remark this answers is not in the document]
Comment (harris-irfan) [reply lms/djangoapps/philu_api/views.py:219]: https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines

Please also refactor in line 228 and line 239

File: lms/djangoapps/philu_api/helpers.py [hunk 2]
```diff
@@ -59,6 +63,15 @@ def get_social_sharing_urls(course_url, meta_tags, tweet_text=None):
...
 def _compile_social_sharing_url(share_url, course_url, url_param, utm_source, text=None):
+    """
+    Create social sharing url by combining url and parameters
+    @param share_url: social platform url
+    @param course_url: course url
+    @param url_param: url parameters
+    @param utm_source: social platform source i.e. Facebook, LinkedIn, Twitter
+    @param text: url text parameter
+    @return: social sharing url
+    """
```
Comment (harris-irfan) [inline lms/djangoapps/philu_api/helpers.py:74]: Aren't we following Google Style docstrings? They are like:

    """
    Summary line.

    Extended description of function.

    Arguments:
        arg1 (int): Description of arg1
        arg2 (str): Description of arg2

    Returns:
        bool: Description of return value
    """

https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#docstrings

https://arbisoft.slack.com/archives/CGYKGH4P5/p1597833417031500

**Review by harris-irfan — CHANGES_REQUESTED** [review]

File: lms/djangoapps/philu_api/helpers.py (hunk 2 above)
Comment (monofal) [reply lms/djangoapps/philu_api/helpers.py:74]: As discussed, this style is recommended not enforced. edX doesn't enforce this style in their code base as well.

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Review by harris-irfan — APPROVED** [review]
LGTM 👍

**Review by mabdullahz — APPROVED** [review]

**Pull request merged**